src/metabase/automagic_dashboards/core.clj

```diff
@@ -1032,7 +1042,7 @@
   [metric opts]
   (automagic-dashboard (merge (->root metric) opts)))
 
-(defn- collect-metrics
+(s/defn ^:private collect-metrics :- (s/maybe [(type Metric)])
```

Comment:
It seems like the calling code expects this to return a sequence of Metrics. So this is to catch bugs sooner and document that fact.